### PR TITLE
Feature/matching endpoint

### PR DESCRIPTION
**Pull request by h-pyo**
Added "preferences" to the Profile schema, which stores an array of IDs for different mentees that the user selected in order on the matching page. Added this mutation call to the matching/index page to edit user's preferences in the db after submitting form.

### CODE REVIEW

**Review by ShawnAviles — APPROVED** [review]
lgtm, I defer to @ulrokx if he prefers for this to be done differently.

File: src/server/api/routers/user.ts
Comment (ulrokx) [inline src/server/api/routers/user.ts:60]: We should probably check to make sure each id is valid i.e. is a mentor/mentee and exists. Or else Patrick Hill will come and find you in your sleep.

And so we know that the entries are for sure valid (although it will probably be prudent to check in the algorithm part as well incase of deleted users)

**Review by ulrokx — APPROVED** [review]
Just a wee bit of input validation

**Pull request merged**
